javascript/tables/src/validateTable.ts

```diff
@@ -0,0 +1,288 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type {
+	CodebookSchema,
+	ColumnError,
+	Field,
+	ValidationResult,
+} from '@datashaper/schema'
+import { DataType } from '@datashaper/schema'
+import type ColumnTable from 'arquero/dist/types/table/column-table'
+
+export function validateTable(
+	table: ColumnTable,
+	codebook: CodebookSchema,
+): ValidationResult {
+	const validationResult: ValidationResult = {
+		columnErrors: [],
+	}
+
+	codebook.fields.forEach((field: Field) => {
+		const columnValues = table.array(field.name)
+		let flag = false
```

Comment:
I'm not sure what the best approach is with `flag` here.

- Right now you only get one error per column, because once `flag === true` you skip the rest of the checks.
- If you remove the flag, you can get a list of errors for the column, which could be helpful - at the expense of more perf required to check everything.

I think I am leaning toward the second option, because it would provide the user with a lot more information if there are problems with the table.

If you did that, you could also split this function apart quite a bit more like:

```
const requiredResult = checkRequiredConstant(columnValues as string, constraints) // returns a ColumnError or undefined
if (requiredResult) {
  validationResult.columnErrors.push(requiredResult)
}
```

That would move more of the logic into each checking function.